element_test.py

```diff
@@ -0,0 +1,57 @@
+import unittest
+import pytest
+from element import QuoteParagraph, TextBlock, ImgBlock, \
+                    ItalicBlock, UnorderedParagraph, LinkBlock, \
+                    StrikethroughBlock
+
+class TestStringMethods(unittest.TestCase):
+
+    def test_text_render(self):
+
+        h1 = TextBlock("test")
+        self.assertEqual(h1.render(), "test")
+
+    def test_quote_render(self):
+        qp = QuoteParagraph("abc")
+        self.assertEqual(qp.render(), "<blockquote> abc </blockquote>")
+        self.assertEqual(QuoteParagraph("<html> rand tag </html>").render(), "<blockquote> <html> rand tag </html> </blockquote>")
+
+
+    def test_img_render(self):
+        ib = ImgBlock("test image", "https://google.com/imagea")
+        self.assertEqual(ib.render(), "<img src='https://google.com/imagea' alt='test image'> </img>")
+
+@pytest.mark.parametrize("content, html", [
+    ( "testing text", "<em>testing text</em>" ),
+    ( "", "<em></em>")
+])
+def test_italic_render(content, html):
+    ib = ItalicBlock(content)
+    assert html == ib.render()
+
+@pytest.mark.parametrize("content, isStart, isEnd, html", [
+    ( "first item", True, None, "<ul><li>first item</li>" ),
+    ( "second item", None, None, "<li>second item</li>" ),
+    ( "last item", None, True, "<li>last item</li></ul>" )
```

Comment:
Should we consider the nested case for a paragraph element like an unordered list?

Reply:
we should support both ordered list and unordered list.

Reply:
> we should support both ordered list and unordered list.

that's my part

Reply:
right, even though ordered list is my part, but these two styles are highly similar. Let's rename the class to `List` and add "isOrdered" flag to indicate it's ordered or unordered. Then, the start or end could be determined correspondingly